ogr/ogrsf_frmts/shape/ogrshapelayer.cpp

```diff
@@ -1674,6 +1674,17 @@ GIntBig OGRShapeLayer::GetFeatureCount(int bForce)
 
 OGRErr OGRShapeLayer::GetExtent(OGREnvelope *psExtent, int bForce)
 
+{
+    OGREnvelope3D envelope3D;
+    const OGRErr retVal{GetExtent3D(0, &envelope3D, bForce)};
+    psExtent->MinX = envelope3D.MinX;
+    psExtent->MinY = envelope3D.MinY;
+    psExtent->MaxX = envelope3D.MaxX;
+    psExtent->MaxY = envelope3D.MaxY;
```

Comment:
As OGREnvelope3D subclass OGREnvelope, I believe below should just work (hopefully...):
```suggestion
    *psExtent = envelope3D;
```

Reply:
clazy not happy:
![image](https://github.com/OSGeo/gdal/assets/142164/f0d6ee19-11f0-45bf-a021-dd978747f97a)

Reply:
would be nice to add a assign operator indeed